src/main/java/com/softserve/academy/spaced/repetition/domain/Person.java

```diff
@@ -14,9 +22,17 @@
     private Long id;
 
     @Column(name = "first_name")
+    @NotNull(message = NULL_MESSAGE, groups = Request.class)
+    @NotEmpty(message = EMPTY_MESSAGE, groups = Request.class)
+    @Length(min = PERSON_FIELD_MIN_LENGTH, max = PERSON_FIELD_MAX_LENGTH, message = PERSON_FIELD_LENGTH_MESSAGE, groups = Request.class)
+    @Pattern(regexp = SPECIAL_SYMBOLS_PATTERN, message = SPECIAL_SYMBOLS_PATTERN_MESSAGE, groups = Request.class)
     private String firstName;
```

Comment:
Lenght and NotEmpty are Hibernate specific. I would recoment to use instead the one from J2EE API https://docs.oracle.com/javaee/7/api/javax/validation/constraints/package-summary.html

Length can be replaced byhttps://docs.oracle.com/javaee/7/api/javax/validation/constraints/Size.html

There is no need to use NotEmpty (and possibly NotNull, please check) in case the minimal required length is specified.